### PR TITLE
93 delete comment

### LINKED ISSUE #93: delete comment

### PR DESCRIPTION
**Pull request by Nour-Agha99**

### CODE REVIEW

**Review by Shatha-Amin — APPROVED** [review]

**Review by mobatra — APPROVED** [review]

**Review by mobatra — APPROVED** [review]

File: src/controllers/comments/deleteCommentsController.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+const deleteCommentsController = async (
+  req: DeleteCommentRequest,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    if (!req.user) throw new CustomError('Unauthorized', 401);
+
+    const { commentId } = req.body;
```
Comment (karamzomlot) [inline src/controllers/comments/deleteCommentsController.ts]: The same here set it as a param

File: src/controllers/comments/deleteCommentsController.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    await deleteCommentSchema.validateAsync({ commentId });
+
+    const { rows: comment } = await getCommentByIdQuery({ commentId });
+
+    if (!comment.length) {
+      throw new CustomError('NOT FOUND', 404);
+    }
+    if (comment[0].user_id !== req.user?.id) {
+      throw new CustomError('NOT FOUND', 404);
```
Comment (karamzomlot) [inline src/controllers/comments/deleteCommentsController.ts]: Unauthorized not Not Fount

File: src/controllers/comments/deleteCommentsController.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    const { commentId } = req.body;
+    await deleteCommentSchema.validateAsync({ commentId });
+
+    const { rows: comment } = await getCommentByIdQuery({ commentId });
+
+    if (!comment.length) {
+      throw new CustomError('NOT FOUND', 404);
+    }
+    if (comment[0].user_id !== req.user?.id) {
```
Comment (karamzomlot) [inline src/controllers/comments/deleteCommentsController.ts:22]: Allow the admin to delete the comment

File: src/routes/comments.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import { addCommentController, deleteCommentsController, editCommentController, getCommentsController } from '../controllers';
+import checkAuth from '../middlewares';
+
+const commentsRouter: Router = express.Router();
+
+commentsRouter.use(checkAuth);
+commentsRouter.get('/', getCommentsController);
+commentsRouter.post('/', addCommentController);
+commentsRouter.put('/', editCommentController);
+commentsRouter.delete('/', deleteCommentsController);
```
Comment (karamzomlot) [inline src/routes/comments.ts]: set the id as a param

File: src/utils/types/requests/index.ts [hunk 5]
```diff
@@ -51,6 +59,19 @@ interface EditPostRequest extends RequestWithDecoded {
...
+    content: string;
+    postId: number;
+    commentId: number;
+  };
+}
+
+interface DeleteCommentRequest extends RequestWithDecoded {
+  body: {
+    commentId: number;
+  };
```
Comment (karamzomlot) [inline src/utils/types/requests/index.ts]: will be change to param instead of body

**Review by karamzomlot — CHANGES_REQUESTED** [review]

**Review by KhaledAref13 — APPROVED** [review]

**Review by karamzomlot — APPROVED** [review]

**Pull request merged**